selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/git/OpenOnGitHubTest.java

```diff
@@ -0,0 +1,160 @@
+
+  @Inject
+  @Named("github.username")
+  private String gitHubUsername;
+
+  @Inject private org.eclipse.che.selenium.pageobject.git.Git git;
+
+  @BeforeClass
+  public void setUp() throws Exception {
+    testUserPreferencesServiceClient.addGitCommitter(gitHubUsername, productUser.getEmail());
+    Path entryPath =
+        Paths.get(getClass().getResource("/projects/default-spring-project").getPath());
+    testProjectServiceClient.importProject(
+        workspace.getId(), entryPath, PROJECT_WITHOUT_GIT_FOLDER, ProjectTemplates.MAVEN_SPRING);
+
+    testRepo.addContent(entryPath);
+    ide.open(workspace);
+    mainBrowserTabHandle = seleniumWebDriver.getWindowHandle();
+    git.importJavaApp(testRepo.getHtmlUrl(), PROJECT_NAME, BLANK);
+    projectExplorer.waitItem(PROJECT_NAME);
+    projectExplorer.quickExpandWithJavaScript();
+  }
+
+  @AfterMethod
+  public void returnToMainWindow() {
+    if (seleniumWebDriver.getWindowHandles().size() > 1) {
+      seleniumWebDriverHelper.closeCurrentWinAndReturnToMainTab(mainBrowserTabHandle);
+    }
+  }
+
+  @Test
+  public void openProjectOnGitHubTest() {
+    projectExplorer.openContextMenuByPathSelectedItem(PROJECT_NAME);
+    projectExplorer.waitContextMenu();
+    projectExplorer.clickOnItemInContextMenu(
+        TestProjectExplorerContextMenuConstants.ContextMenuFirstLevelItems.OPEN_ON_GITHUB);
+    seleniumWebDriverHelper.switchToNextWindow(seleniumWebDriver.getWindowHandle());
+    Assert.assertTrue(seleniumWebDriver.getCurrentUrl().startsWith(testRepo.getHtmlUrl()));
+  }
+
+  @Test
+  public void openFolderOnGitHubTest() {
+    projectExplorer.waitAndSelectItem(PROJECT_NAME + PATH_TO_EXPAND);
+    projectExplorer.openContextMenuByPathSelectedItem(PROJECT_NAME + PATH_TO_EXPAND);
+    projectExplorer.waitContextMenu();
+    projectExplorer.clickOnItemInContextMenu(
+        TestProjectExplorerContextMenuConstants.ContextMenuFirstLevelItems.OPEN_ON_GITHUB);
+    seleniumWebDriverHelper.switchToNextWindow(seleniumWebDriver.getWindowHandle());
+    Assert.assertEquals(
+        seleniumWebDriver.getCurrentUrl(), testRepo.getHtmlUrl() + "/tree/master" + PATH_TO_EXPAND);
+  }
+
+  @Test
+  public void openFileOnGitHubTest() {
+    projectExplorer.openItemByPath(PROJECT_NAME + "/src/main/java/che/eclipse/sample/Aclass.java");
+    editor.selectLines(14, 8);
+    editor.openContextMenuInEditor();
+    editor.clickOnItemInContextMenu(OPEN_ON_GITHUB);
+    seleniumWebDriverHelper.switchToNextWindow(seleniumWebDriver.getWindowHandle());
+    Assert.assertEquals(
+        seleniumWebDriver.getCurrentUrl(),
+        testRepo.getHtmlUrl() + "/blob/master" + PATH_TO_EXPAND + "/Aclass.java#L14-L16");
+  }
+
+  @Test
+  public void checkProjectWithoutGitFolder() {
+    projectExplorer.openContextMenuByPathSelectedItem(PROJECT_WITHOUT_GIT_FOLDER);
+    projectExplorer.waitItemOnContexMenuIsNotVisible(
+        TestProjectExplorerContextMenuConstants.ContextMenuFirstLevelItems.OPEN_ON_GITHUB);
```

Comment:
Static import?

Reply:
In this case is not applicable because we will get namespace conflict